execution/function/operator.go

```diff
@@ -33,6 +33,96 @@ type functionOperator struct {
 	pointBuf     []promql.Point
 }
 
+type absentOperator struct {
+	mint        int64
+	maxt        int64
+	step        int64
+	currentStep int64
+	stepsBatch  int
+	funcExpr    *parser.Call
+	series      []labels.Labels
+	once        sync.Once
+
+	next       model.VectorOperator
+	vectorPool *model.VectorPool
+}
+
+func (o *absentOperator) Explain() (me string, next []model.VectorOperator) {
+	return fmt.Sprintf("[*absentOperator] %v()", o.funcExpr.Func.Name), []model.VectorOperator{}
+}
+
+func (o *absentOperator) Series(_ context.Context) ([]labels.Labels, error) {
+	o.loadSeries()
+
+	return o.series, nil
+}
+
+func (o *absentOperator) loadSeries() {
+	o.once.Do(func() {
+		// https://github.com/prometheus/prometheus/blob/main/promql/functions.go#L1385
+		var lm []*labels.Matcher
+		switch n := o.funcExpr.Args[0].(type) {
+		case *parser.VectorSelector:
+			lm = n.LabelMatchers
+		case *parser.MatrixSelector:
+			lm = n.VectorSelector.(*parser.VectorSelector).LabelMatchers
+		default:
+			o.series = []labels.Labels{labels.EmptyLabels()}
+			return
+		}
```

Comment:
The argument itself is the operator next IIUC as it can be nested expressions.

Reply:
Thats a good comment! I tested it out briefly but that would fail `absent(absent(exists{foo="bar"}))`; in this case prometheus returns `{} 1` but if i use nested labels it would be `{foo="bar"} 1` ; 

```
                    	exp: &promql.Result{Err:error(nil), Value:promql.Vector{promql.Sample{Point:promql.Point{T:50000, V:1, H:(*histogram.FloatHistogram)(nil)}, Metric:labels.Labels{}}}, Warnings:storage.Warnings(nil)}

                    	got: &promql.Result{Err:error(nil), Value:promql.Vector{promql.Sample{Point:promql.Point{T:50000, V:1, H:(*histogram.FloatHistogram)(nil)}, Metric:labels.Labels{labels.Label{Name:"job", Value:"myjob"}}}}, Warnings:storage.Warnings(nil)}
```
let me actually push that test.

Reply:
Or did i misunderstand something? This is relatively new to me

Reply:
I haven't tested this myself but in theory, the inner `absent(exists{foo="bar"})` will give empty series if the series exists. So the outer `absent` will return `{}`. Probably there is edge case in the new engine implementation?

Reply:
Ah i forgot to say: I baseically reused upstream [promql implementation](https://github.com/prometheus/prometheus/blob/666f61a4d5d57e2a4ea2adbd27a3dc25c599e132/promql/functions.go#L1385) for the labels logic. If the expression is not a selector it will use empty labels every time. So we should ignore `next` here.